src/utils/notifyWithTost.js

```diff
@@ -48,9 +48,21 @@ const messageTemplates = {
             render({ data }) {
                 console.log({ data });
                 if (data.code === 4001) return "User denied transaction";
+                if (data?.error?.code === -32603) return <TxRevertMessage errorMessage={data.error.message} />;
                 return "Failed to execute transaction";
             },
         },
     },
     // add more message types here as needed
 };
+
+const TxRevertMessage = ({ errorMessage }) => {
```

Comment:
Could we generalize this so it works for success messages as well? And do you it would be good?

Reply:
Certainly! There is a lot of room for improving this layer. I was going to address your suggestion in #128  issue